src/charmory/tools/huggingface_xview_images_reduced.ipynb

```diff
@@ -0,0 +1,1087 @@
+       "unique_id\n",
+       "1    3555\n",
+       "0    1154\n",
+       "3     111\n",
+       "2      18\n",
+       "4      11\n",
+       "5       5\n",
+       "Name: count, dtype: int64"
+      ]
+     },
+     "execution_count": 30,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df.unique_id.value_counts()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 35,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "new_cat\n",
+       "Building               3555\n",
+       "Passenger Vehicle      1154\n",
+       "Truck                   111\n",
+       "Other                    18\n",
+       "Engineering Vehicle      11\n",
+       "Maritime Vessel           5\n",
+       "Name: count, dtype: int64"
+      ]
+     },
+     "execution_count": 35,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "df.new_cat.value_counts()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "finished with 20\n",
+      "finished with 41\n",
+      "finished with 62\n",
+      "finished with 83\n"
+     ]
+    }
+   ],
+   "source": [
+    "df['shape'] = 0\n",
+    "df_final = pd.DataFrame()\n",
+    "LIST =[]; i = 0\n",
+    "for group_name, df2 in df.groupby('image_id_file'):\n",
+    "    #df2 = df2.drop(['level_0'],axis=1)\n",
+    "    df2 = df2.reset_index()\n",
+    "    df_append = pd.DataFrame(index=range(1),columns=['image_id','image','width','height','objects'])\n",
+    "    \n",
+    "    height, width, _ = np.array(Image.open('basic/'+str(group_name))).shape\n",
+    "    df_append.at[0,'image_id'] = df2['image_id_file'][0]\n",
+    "    df_append.at[0,'image'] = 'basic/'+str(group_name)\n",
+    "    df_append.at[0,'width'] = width\n",
+    "    df_append.at[0,'height'] = height\n",
+    "    df_append.at[0,'objects'] = dict({\n",
+    "        'id': df2['index'].tolist(),\n",
+    "        'area': df2['area'].tolist(),\n",
+    "        'bbox': df2['list_of_cooridates'].tolist(),\n",
+    "        'category': df2['unique_id'].tolist()\n",
+    "    }\n",
+    "    )\n",
+    "    \n",
+    "    LIST.append(df_append)\n",
+    "    if len(LIST) > 20:\n",
+    "        df_concat = pd.concat(LIST)\n",
+    "        df_final = pd.concat([df_final,df_concat])\n",
+    "        LIST = []\n",
+    "        print('finished with ' + str(i))\n",
+    "    i += 1\n",
+    "if len(LIST) > 0:\n",
+    "    df_concat = pd.concat(LIST)\n",
+    "    df_final = pd.concat([df_final,df_concat])\n",
+    "    \n",
+    "df = df_final.reset_index()\n",
+    "del df_final, df2,df_concat, df_append"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hg_dataset = Dataset(pa.Table.from_pandas(df))\n",
```

Comment:
Why not use `Dataset.from_pandas` to avoid the intermediate conversion to arrow?

Reply:
Thanks, I changed to that